core/base/src/TDirectory.cxx

```diff
@@ -1028,9 +1028,11 @@ void TDirectory::FillFullPath(TString& buf) const
 ///    gDirectory->cd("b");
 ///    gDirectory->mkdir("d");
 /// ~~~
+/// ifNotExist is just for compatibility with the TDirectoryFile override
 
-TDirectory *TDirectory::mkdir(const char *name, const char *title)
+TDirectory *TDirectory::mkdir(const char *name, const char *title, Bool_t ifNotExist)
 {
+   if (ifNotExist) Error("mkdir","ifNotExist only valid for TDirectoryFile mkdir",name);
```

Comment:
```suggestion
   if (ifNotExist)
      Error("mkdir", "ifNotExist only valid for TDirectoryFile mkdir", name);
```
(more compliant with coding conventions).

Reply:
Why wouldn't the behavior of TDirectory and TDirectoryFile in regard to the new parameter be the same?